### PR TITLE
Handle connection loss

### PR DESCRIPTION
**Pull request by lucksus**
This fixes several issues that all occur when a peer is lost and that were rendering the running instance broken through an unhandled exception and/or loosing all information about the peer so that reconnecting was not possible.

Mainly this line in `dht-fullsync.js` was a logical error I believe and gets removed here:
```js
// Remove peer from peerStore
this.dropPeerLocal(ref.peerAddress)
```
Peer store is holding the information about the peer such as the transport address and not dropping it results in `_ensureCIdForPeer()` being able to reconnect again.

### CODE REVIEW

File: lib/n3h-mod-connection-wss/wss-connection.js [hunk 1]
```diff
@@ -55,7 +55,14 @@ class ConnectionBackendWss extends AsyncClass {
           ws.close(1001, 'stale connection')
           await this._handleClose(id)
         } else if (Date.now() - ws._$_lastMsg >= 2000) {
-          ws.ping()
+          try {
+            ws.ping()
```
Comment (neonphog) [inline lib/n3h-mod-connection-wss/wss-connection.js]: @lucksus - it looks like this `ping()` and the `send()` below use callbacks for asynchronous behavior (https://github.com/websockets/ws/blob/HEAD/doc/ws.md#websocketpingdata-mask-callback), so the try/catches may not catch errors.

promisify / await:
```javascript
const { $p } = require('../n3h-common')
await $p(cb => { ws.ping(cb) })
```

File: lib/hackmode/p2p-backend-hackmode-peer.js [hunk 2]
```diff
@@ -302,8 +302,14 @@ class P2pBackendHackmodePeer extends AsyncClass {
 
     switch (uri.protocol) {
       case 'wss:':
-        return this._ensureConnection(
-          peerInfo.peerTransport + '?a=' + peerInfo.peerAddress)
+        if (peerInfo.peerAddress === this.getId()) {
+          // Don't try to connect self
+          return Promise.resolve()
```
Comment (neonphog) [inline lib/hackmode/p2p-backend-hackmode-peer.js]: - async functions wrap returns in promises, so you can just `return`
- the caller is expecting a cid to be returned, I think we might need to do this check elsewhere.

File: lib/n3h-mod-connection-wss/wss-connection.js (hunk 1 above)
Comment (lucksus) [reply lib/n3h-mod-connection-wss/wss-connection.js]: I just followed the stacktrace that was printed out to these lines. Adding this try catch fixes the problem.

**Comment by neonphog** [comment]
merged into https://github.com/holochain/n3h/pull/96

**Pull request closed without merging**